pkg/s3select/select.go

```diff
@@ -276,91 +297,111 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
 					}
 				}
 			}
+
 			filtrCount++
 		}
 		counter++
 	}
+
+}
+
+//ReadIndividualRecord return a map[string]interfaces{} for each row in csv and each json for JSON document
+func readIndividualRecord(inputType SelectQuery) (map[string]interface{}, error) {
+	var individualRecord map[string]interface{}
+	var err error
+	switch inputType.(type) {
+	case *CSVInput:
+		individualRecord, err = inputType.Read()
+	case *JSONInput:
+		individualRecord, err = inputType.Read()
+	}
+	// In case an exception arises while parsing
+	if err != nil {
+		return nil, err
+	}
+	return individualRecord, nil
 }
 
 // printAsterix helps to print out the entire row if an asterix is used.
-func (reader *Input) printAsterix(record []string) string {
+func printAsterix(record []string, inputType SelectQuery) string {
 	myRow := record[0]
 	for i := 1; i < len(record); i++ {
-		myRow = myRow + reader.options.OutputFieldDelimiter + record[i]
+		myRow = myRow + inputType.getOutPutFieldDelimitter() + record[i]
```

Comment:
typo `getOutPutFieldDelimitter` should be `getOutputFieldDelimitter`

Reply:
Done

Reply:
Sorry, missed one more type `getOutputFieldDelimitter` should be `getOutputFieldDelimiter`